### PR TITLE
chore(cli): add environment variable to disable version checks

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
This env var could be helpful in case of bugs in the version detection

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       image: docker.mirror.hashicorp.services/hashicorp/jsii-terraform
     env:
       CHECKPOINT_DISABLE: "1"
-
+      DISABLE_VERSION_CHECK: "1"
```
Comment (ansgarm) [inline .github/workflows/release.yml:22]: While this unblocks us in the short term, I'd rather not do this and try to figure out and fix the underlying issue.

File: website/docs/cdktf/create-and-deploy/environment-variables.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## CDKTF_LOG_LEVEL
+
+Can be set to one of the following values:
+
+- `debug`
+- `info`
+- `warn`
+- `error`
+- `fatal`
```
Comment (ansgarm) [inline website/docs/cdktf/create-and-deploy/environment-variables.mdx:36]: It also supports `all`, right?

File: website/docs/cdktf/create-and-deploy/environment-variables.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+If set this token will be used to authenticate against Terraform Cloud, it needs to be set to a valid [Terraform Cloud API token](/cloud-docs/users-teams-organizations/api-tokens).
+
+## http_proxy or HTTP_PROXY
+
+If set this proxy will be used for all HTTP requests.
+
+## DISABLE_VERSION_CHECK
+
+If set to a not-empty value, no version check will be performed. If it is not set we will verify your installed library version matches the CLI version and warn if they do not match.
```
Comment (ansgarm) [inline website/docs/cdktf/create-and-deploy/environment-variables.mdx:58]:
```suggestion
If set to a non-empty value, no version check will be performed. If it is not set we will verify your installed library version matches the CLI version and warn if they do not match.
```

File: packages/cdktf-cli/bin/cmds/helper/check-environment.ts [hunk 4]
```diff
@@ -55,24 +55,26 @@ export async function checkEnvironment() {
 }
 
 export async function verifySimilarLibraryVersion() {
+  if (process.env.DISABLE_VERSION_CHECK) {
+    logger.debug("Version check disabled via Environment variable");
```
Comment (ansgarm) [inline packages/cdktf-cli/bin/cmds/helper/check-environment.ts:59]:
```suggestion
    logger.debug("Version check disabled via environment variable");
```

File: packages/cdktf-cli/bin/cmds/helper/check-environment.ts [hunk 5]
```diff
@@ -55,24 +55,26 @@ export async function checkEnvironment() {
 }
 
 export async function verifySimilarLibraryVersion() {
+  if (process.env.DISABLE_VERSION_CHECK) {
```
Comment (ansgarm) [inline packages/cdktf-cli/bin/cmds/helper/check-environment.ts:58]: Should we allow `"false"` to not disable the version check?
```suggestion
  if (process.env.DISABLE_VERSION_CHECK && process.env.DISABLE_VERSION_CHECK !== "false") {
```

**Review by ansgarm — APPROVED** [review]

File: packages/cdktf-cli/bin/cmds/helper/check-environment.ts (hunk 5 above)
Comment (DanielMSchmidt) [reply packages/cdktf-cli/bin/cmds/helper/check-environment.ts:58]: I read through every one of our `process.env` calls today and none have a check like this. We have one for logging that checks if it's === false. I believe we should overhaul what we currently have and check if it's set to 1 or true, but I'd rather do it globally after this merged

Comment (ansgarm) [reply packages/cdktf-cli/bin/cmds/helper/check-environment.ts:58]: fine with me

**Pull request merged**